crates/iroha_data_model/src/isi.rs

```diff
@@ -939,14 +939,14 @@ mod transparent {
         pub fn new(trigger: TriggerId) -> Self {
             Self {
                 trigger,
-                args: None,
+                args: JsonString::default(),
             }
         }
 
         /// Add trigger execution args
         #[must_use]
-        pub fn with_args<T: serde::Serialize>(mut self, args: &T) -> Self {
-            self.args = Some(JsonString::new(args));
+        pub fn with_args(mut self, args: impl Into<JsonString>) -> Self {
```

Comment:
It is unfortunate that the user will have to implement these for the args types. Is there a reason you decided against `T: serde::Serialize` bound?

Reply:
reverted